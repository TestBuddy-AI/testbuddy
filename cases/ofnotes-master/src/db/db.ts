import Dexie from "dexie";
import initialNotes from "./initialNotes";
import slugify from "slugify";

export const LIST_SIZE = 100;

export interface UserNoteFields {
  title: string;
  tags: string[];
  md: string;
}

export interface AutoNoteFields {
  modified: number;
  slug: string;
  id?: number;
}

export interface Note extends UserNoteFields, AutoNoteFields {}

export interface UnsavedNote extends UserNoteFields, Partial<AutoNoteFields> {}

export class NotesDB extends Dexie {
  notes: Dexie.Table<Note, number>;

  constructor() {
    super("NotesDB");

    this.version(1).stores({
      notes: "++id, &title, *tags, modified, &slug",
    });
    this.notes = this.table("notes");

    this.on("populate", () => {
      this.notes.bulkAdd(initialNotes);
    });
  }

  async get(slug: string) {
    return this.notes.get({ slug });
  }

  async create(note: UserNoteFields) {
    const id = await this.notes.add({
      ...(await this.validate(note)),
      slug: slugify(note.title, { lower: true }),
      modified: Date.now(),
      tags: note.tags.filter((t) => Boolean(t)),
    });
    return (await this.notes.get(id)) as Note;
  }

  async list(start: number = 0) {
    return this.notes
      .orderBy("modified")
      .reverse()
      .offset(start)
      .limit(LIST_SIZE)
      .toArray();
  }

  async listWithFilter(filter: string, start: number = 0) {
    return this.notes
      .where("title")
      .startsWithIgnoreCase(filter)
      .or("tags")
      .startsWithIgnoreCase(filter)
      .offset(start)
      .limit(LIST_SIZE)
      .toArray();
  }

  async update(noteId: number, note: UserNoteFields) {
    if (!noteId) {
      throw new Error("invalid note id");
    }
    await this.notes.update(noteId, {
      ...(await this.validate(note)),
      slug: slugify(note.title, { lower: true }),
      modified: Date.now(),
      tags: note.tags.filter((t) => Boolean(t)),
    });
    return (await this.notes.get(noteId)) as Note;
  }

  async remove(noteId: number) {
    return await this.notes.delete(noteId);
  }

  async validate(note: UserNoteFields) {
    if (!note.title) {
      throw new Error("title is required");
    }

    return note;
  }
}

export const db = new NotesDB();
