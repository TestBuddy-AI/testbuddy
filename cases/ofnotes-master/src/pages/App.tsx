import React from "react";
import { Route, Switch } from "react-router-dom";
import { Layout } from "components";
import { Home, Note, Form } from "pages";
import {
  ProvideThemeContext,
  ProvidSideBarContext,
  ProvideNoteContext,
} from "hooks";
import moment from "moment";
import { createBrowserHistory } from "history";
import { Router } from "react-router-dom";
import ReactGA from "react-ga";

const history = createBrowserHistory();

if (process.env.NODE_ENV === "production") {
  ReactGA.initialize("UA-164169628-1");
  ReactGA.pageview(window.location.pathname + window.location.search);
  history.listen((location) => {
    ReactGA.set({ page: location.pathname });
    ReactGA.pageview(location.pathname);
  });
}

moment.updateLocale("en", {
  relativeTime: {
    s: "1s",
    ss: "%ds",
    m: "1m",
    mm: "%dm",
    h: "1h",
    hh: "%dh",
    d: "1d",
    dd: "%dd",
    y: "1y",
    yy: "%dy",
  },
});

export const App = () => (
  <Router history={history}>
    <ProvideNoteContext>
      <ProvideThemeContext>
        <ProvidSideBarContext>
          <Layout>
            <Switch>
              <Route exact path="/" component={Home} />
              <Route exact path="/create" component={Form} />
              <Route exact path="/:slug/edit" component={Form} />
              <Route exact path="/:slug" component={Note} />
            </Switch>
          </Layout>
        </ProvidSideBarContext>
      </ProvideThemeContext>
    </ProvideNoteContext>
  </Router>
);
