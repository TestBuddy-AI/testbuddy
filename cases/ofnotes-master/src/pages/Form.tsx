import React, { useState } from "react";
import { useParams } from "react-router-dom";
import { useSwipeable } from "react-swipeable";
import {
  Tabs,
  Tab,
  makeStyles,
  createStyles,
  Divider,
  useTheme,
  useMediaQuery,
} from "@material-ui/core";
import {
  NoteForm,
  MainPanel,
  RenderedNote,
  ConfirmDeleteModal,
} from "components";
import { useNoteForm } from "hooks";

const useStyles = makeStyles((theme) =>
  createStyles({
    root: {
      display: "flex",
      flexDirection: "column",
      width: "100%",
      height: "100%",
    },
    tabs: {
      width: "100%",
      height: 48,
      backgroundColor: theme.palette.background.paper,
      boxShadow: theme.shadows[2],
      [theme.breakpoints.up("md")]: {
        backgroundColor: theme.palette.background.default,
        boxShadow: "none",
      },
      [theme.breakpoints.up("lg")]: {
        display: "none",
      },
      "& hr": {
        width: "100%",
        margin: "0 auto",
        [theme.breakpoints.up("md")]: {
          width: "80%",
        },
      },
    },
    content: {
      display: "flex",
      width: "100%",
      height: "100%",
      overflowY: "auto",
      overflowX: "hidden",
    },
  })
);

export const Form: React.FC = () => {
  const classes = useStyles();
  const theme = useTheme();
  const expanded = useMediaQuery(theme.breakpoints.up("lg"));
  const { slug } = useParams();
  const { savedValues, onDelete, ...formState } = useNoteForm(slug);

  const [tab, setTab] = useState(0);
  const [deleteOpen, setDeleteOpen] = useState(false);

  const swipeHandler = useSwipeable({
    onSwipedRight: () => setTab(0),
    onSwipedLeft: () => setTab(1),
  });

  return (
    <div className={classes.root}>
      <div className={classes.tabs}>
        <Tabs value={tab} onChange={(e, val) => setTab(val)} centered>
          <Tab label="Write" />
          <Tab label="Preview" />
        </Tabs>
        <Divider />
      </div>
      <div
        className={classes.content}
        data-testid="form-container"
        {...swipeHandler}
      >
        {(expanded || tab === 0) && (
          <MainPanel>
            <NoteForm
              {...formState}
              showDelete={Boolean(savedValues?.id)}
              onDelete={() => setDeleteOpen(true)}
            />
          </MainPanel>
        )}
        {(expanded || tab === 1) && (
          <MainPanel>
            <RenderedNote note={savedValues} />
          </MainPanel>
        )}
      </div>
      <ConfirmDeleteModal
        open={deleteOpen}
        title={`Permanently delete ${savedValues?.title}?`}
        onClose={() => setDeleteOpen(false)}
        onConfirm={() => onDelete()}
      />
    </div>
  );
};
