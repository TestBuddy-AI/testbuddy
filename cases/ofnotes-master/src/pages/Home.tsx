import React from "react";
import { MainPanel, CreateNoteFAB, RenderedNote } from "components";
import { makeStyles, createStyles, Typography } from "@material-ui/core";
import { useNote } from "hooks";

const useStyles = makeStyles((theme) =>
  createStyles({
    content: {
      height: "100%",
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
      alignItems: "center",

      "& h4": {
        textAlign: "center",
      },
    },
  })
);

export const Home: React.FC = () => {
  const classes = useStyles();
  const { note, loading } = useNote("welcome");

  return (
    <MainPanel>
      {note ? (
        <RenderedNote note={note} />
      ) : !loading ? (
        <div className={classes.content}>
          <Typography variant="h4" color="textSecondary">
            Select a note to get started
          </Typography>
        </div>
      ) : (
        ""
      )}
      <CreateNoteFAB />
    </MainPanel>
  );
};
