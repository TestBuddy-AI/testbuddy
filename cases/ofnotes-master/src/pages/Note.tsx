import React from "react";
import { useParams } from "react-router-dom";
import { makeStyles, createStyles, Typography } from "@material-ui/core";
import { MainPanel, RenderedNote, CreateNoteFAB } from "components";
import { useNote } from "hooks";

const useStyles = makeStyles((theme) =>
  createStyles({
    content: {
      height: "100%",
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
      alignItems: "center",

      "& h4": {
        textAlign: "center",
      },
    },
  })
);

export const Note: React.FC = () => {
  const classes = useStyles();
  const { slug } = useParams();
  const { note, loading } = useNote(slug ?? "");

  return (
    <MainPanel>
      {note ? (
        <RenderedNote note={note} />
      ) : !loading ? (
        <div className={classes.content}>
          <Typography variant="h4" color="textSecondary">
            Not Found
          </Typography>
        </div>
      ) : (
        ""
      )}
      <CreateNoteFAB />
    </MainPanel>
  );
};
