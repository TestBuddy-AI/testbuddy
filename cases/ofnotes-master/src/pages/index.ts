export * from "./App";
export * from "./Home";
export * from "./Note";
export * from "./Form";
