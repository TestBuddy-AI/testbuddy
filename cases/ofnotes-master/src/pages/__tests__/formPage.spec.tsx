import React from "react";
import { render, fakeNote, fireEvent, wait, mocked } from "test-utils";
import { useMediaQuery } from "@material-ui/core";
import { Form } from "pages";
import * as router from "react-router-dom";
import { Note } from "db/db";

jest.mock("@material-ui/core/useMediaQuery");
const mockMediaQuery = mocked(useMediaQuery);

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: jest.fn(),
}));
const mockRouter = mocked(router);

const mockNoteCtx = {
  getNote: jest.fn(),
  updateNote: jest.fn(),
  deleteNote: jest.fn(),
  createNote: jest.fn(),
};

jest.useFakeTimers();

afterEach(() => {
  mockRouter.useParams.mockReset();
  mockMediaQuery.mockReset();
  Object.values(mockNoteCtx).forEach((fn) => fn.mockReset());
});

afterAll(() => {
  jest.restoreAllMocks();
});

interface RenderEditing {
  fullWidth?: boolean;
  note?: Note;
}

const renderPage = async ({ fullWidth = true, note }: RenderEditing = {}) => {
  mockMediaQuery.mockReturnValue(fullWidth);
  mockRouter.useParams.mockReturnValue(note ? { slug: note.slug } : {});
  mockNoteCtx.getNote.mockResolvedValue(note);
  const { queryByLabelText, ...rest } = render(<Form />, {
    noteCtx: mockNoteCtx,
  });
  await wait(() => {
    expect(mockMediaQuery).toHaveBeenCalled();
    expect(mockRouter.useParams).toHaveBeenCalled();
    expect(queryByLabelText("Title")).toBeInTheDocument();
  });
  return { queryByLabelText, ...rest };
};

const renderEditing = async ({
  note = fakeNote,
  fullWidth = true,
}: RenderEditing = {}) => {
  const { queryByLabelText, ...rest } = await renderPage({ note, fullWidth });
  await wait(() => {
    expect(mockNoteCtx.getNote).toHaveBeenCalledTimes(1);
    expect(mockNoteCtx.getNote.mock.calls[0][0]).toEqual(note.slug);
    expect(queryByLabelText("Title")).toHaveValue(note.title);
  });
  return { queryByLabelText, ...rest };
};

const renderCreating = async ({ fullWidth = true }: RenderEditing = {}) => {
  const { queryByLabelText, ...rest } = await renderPage({
    note: undefined,
    fullWidth,
  });
  await wait(() => {
    expect(mockNoteCtx.getNote).not.toHaveBeenCalled();
    expect(queryByLabelText("Title")).toHaveValue("");
  });
  return { queryByLabelText, ...rest };
};

describe("Form page navigation", () => {
  it("matches snapshot", async () => {
    const { asFragment } = await renderEditing();
    expect(asFragment()).toMatchSnapshot();
  });

  it("can switch tabs by clicking", async () => {
    const { getByText, queryByLabelText, queryByText } = await renderEditing({
      fullWidth: false,
    });
    await wait(() => {
      expect(queryByText(fakeNote.title)).not.toBeInTheDocument();
    });
    fireEvent.click(getByText("Preview"));
    await wait(() => {
      expect(queryByLabelText("Title")).not.toBeInTheDocument();
      expect(queryByText(fakeNote.title)).toBeInTheDocument();
    });
  });

  it("can switch tabs by swiping", async () => {
    const { getByTestId, queryByLabelText, queryByText } = await renderEditing({
      fullWidth: false,
    });
    await wait(() => {
      expect(queryByText(fakeNote.title)).not.toBeInTheDocument();
    });
    const container = getByTestId("form-container");

    // swipe right
    fireEvent.touchStart(container, {
      touches: [{ clientX: 25, clientY: 10 }],
    });
    fireEvent.touchMove(container, {
      touches: [{ clientX: 5, clientY: 10 }],
    });
    fireEvent.touchEnd(container);
    await wait(() => {
      expect(queryByLabelText("Title")).not.toBeInTheDocument();
      expect(queryByText(fakeNote.title)).toBeInTheDocument();
    });

    // swipe left
    fireEvent.touchStart(container, {
      touches: [{ clientX: 5, clientY: 10 }],
    });
    fireEvent.touchMove(container, {
      touches: [{ clientX: 25, clientY: 10 }],
    });
    fireEvent.touchEnd(container);
    await wait(() => {
      expect(queryByLabelText("Title")).toBeInTheDocument();
      expect(queryByText(fakeNote.title)).not.toBeInTheDocument();
    });
  });

  it("can open and close delete modal", async () => {
    const { getByText, container } = await renderEditing();
    await wait(() => {
      expect(getByText(fakeNote.title)).toBeInTheDocument();
      expect(getByText("Delete")).toBeInTheDocument();
    });
    fireEvent.click(getByText(/Delete/));
    await wait(() => {
      expect(container.querySelector(".MuiDialog-paper")).toBeInTheDocument();
    });
    fireEvent.click(getByText("Cancel"));
    // TODO: find a way to do this assertion
    await wait(() => {
      expect(container.querySelector(".MuiDialog-paper")).toBeInTheDocument();
    });
  });
});

describe("Form Page edit note", () => {
  it("fills form based on editing note", async () => {
    const { getByLabelText } = await renderEditing();
    await wait(() => {
      expect(getByLabelText("Title")).toHaveValue(fakeNote.title);
      expect(getByLabelText("Note")).toHaveValue(fakeNote.md);
      expect(getByLabelText("Tags")).toHaveValue(fakeNote.tags.join(", "));
    });
  });

  it("renders markdown for editing note", async () => {
    const note = {
      ...fakeNote,
      md: "# test h1",
    };
    const { getByText } = await renderEditing({ note });
    await wait(() => {
      expect(getByText(fakeNote.title)).toBeInTheDocument();
      expect(getByText("test h1")).toBeInTheDocument();
      expect(getByText("test h1")).toHaveClass("MuiTypography-root");
      expect(getByText("test h1")).toHaveAttribute("id", "test-h1");
    });
  });

  it("can delete note", async () => {
    const { getByText, getAllByText } = await renderEditing();
    fireEvent.click(getByText(/Delete/));
    await wait(() => {
      expect(
        getByText(`Permanently delete ${fakeNote.title}?`)
      ).toBeInTheDocument();
    });
    fireEvent.click(getAllByText("Delete")[1]);
    await wait(() => {
      expect(mockNoteCtx.deleteNote).toHaveBeenCalled();
      expect(mockNoteCtx.deleteNote.mock.calls[0][0]).toEqual(fakeNote.id);
    });
  });

  it("can update note title", async () => {
    mockNoteCtx.updateNote.mockResolvedValue({
      ...fakeNote,
      title: "updated title",
    });
    const { getByLabelText, getByText } = await renderEditing();
    await wait(() => {
      expect(getByLabelText("Title")).toHaveValue(fakeNote.title);
    });
    fireEvent.change(getByLabelText("Title"), {
      target: { value: "updated title" },
    });
    await wait(() => {
      jest.runAllTimers();
      expect(getByLabelText("Title")).toHaveValue("updated title");
      expect(mockNoteCtx.updateNote).toHaveBeenCalled();
      expect(mockNoteCtx.updateNote.mock.calls[0][0]).toEqual(fakeNote.id);
      expect(mockNoteCtx.updateNote.mock.calls[0][1]).toMatchObject({
        title: "updated title",
      });
      expect(getByText("updated title")).toBeInTheDocument();
    });
  });

  it("update title handles uniqueness error", async () => {
    const err = new Error();
    Object.assign(err, {
      failures: [{ name: "ConstraintError" }],
    });
    mockNoteCtx.updateNote.mockRejectedValue(err);
    const { getByLabelText, getByText } = await renderEditing();
    await wait(() => {
      expect(getByLabelText("Title")).toHaveValue(fakeNote.title);
    });
    fireEvent.change(getByLabelText("Title"), {
      target: { value: "duplicate title" },
    });
    await wait(() => {
      jest.runAllTimers();
      expect(getByLabelText("Title")).toHaveValue("duplicate title");
      expect(mockNoteCtx.updateNote).toHaveBeenCalled();
      expect(mockNoteCtx.updateNote.mock.calls[0][0]).toEqual(fakeNote.id);
      expect(mockNoteCtx.updateNote.mock.calls[0][1]).toMatchObject({
        title: "duplicate title",
      });
      expect(getByText("Title must be unique")).toBeInTheDocument();
    });
  });

  it("update handles error with a message set", async () => {
    const err = new Error();
    err.message = "test other error message";
    mockNoteCtx.updateNote.mockRejectedValue(err);
    const { getByLabelText, getByText } = await renderEditing();
    await wait(() => {
      expect(getByLabelText("Title")).toHaveValue(fakeNote.title);
    });
    fireEvent.change(getByLabelText("Title"), {
      target: { value: "new title" },
    });
    await wait(() => {
      jest.runAllTimers();
      expect(getByLabelText("Title")).toHaveValue("new title");
      expect(mockNoteCtx.updateNote).toHaveBeenCalled();
      expect(getByText("test other error message")).toBeInTheDocument();
    });
  });

  it("update handles other errors", async () => {
    const err = new Error();
    mockNoteCtx.updateNote.mockRejectedValue(err);
    const { getByLabelText, getByText } = await renderEditing();
    await wait(() => {
      expect(getByLabelText("Title")).toHaveValue(fakeNote.title);
    });
    fireEvent.change(getByLabelText("Title"), {
      target: { value: "other title" },
    });
    await wait(() => {
      expect(getByLabelText("Title")).toHaveValue("other title");
    });
    await wait(() => {
      jest.runAllTimers();
      expect(mockNoteCtx.updateNote).toHaveBeenCalled();
      expect(
        getByText("Something went wrong updating this note")
      ).toBeInTheDocument();
    });
  }, 10000);

  it("can update note markdown", async () => {
    mockNoteCtx.updateNote.mockResolvedValue({
      ...fakeNote,
      md: "# test h2",
    });
    const { getByLabelText, getByText } = await renderEditing();
    await wait(() => {
      expect(getByLabelText("Note")).toHaveValue(fakeNote.md);
    });
    fireEvent.change(getByLabelText("Note"), {
      target: { value: "## test h2" },
    });
    await wait(() => {
      jest.runAllTimers();
      expect(getByLabelText("Note")).toHaveValue("## test h2");
      expect(mockNoteCtx.updateNote).toHaveBeenCalled();
      expect(mockNoteCtx.updateNote.mock.calls[0][0]).toEqual(fakeNote.id);
      expect(mockNoteCtx.updateNote.mock.calls[0][1]).toMatchObject({
        md: "## test h2",
      });
      expect(getByText("test h2")).toBeInTheDocument();
    });
  });

  it("can update note tags", async () => {
    mockNoteCtx.updateNote.mockResolvedValue({
      ...fakeNote,
      tags: ["testing", "is", "awesome"],
    });
    const { getByLabelText, getByText } = await renderEditing();
    await wait(() => {
      expect(getByLabelText("Tags")).toHaveValue(fakeNote.tags.join(", "));
    });
    fireEvent.change(getByLabelText("Tags"), {
      target: { value: "testing is awesome" },
    });
    await wait(() => {
      jest.runAllTimers();
      expect(getByLabelText("Tags")).toHaveValue("testing is awesome");
      expect(mockNoteCtx.updateNote).toHaveBeenCalled();
      expect(mockNoteCtx.updateNote.mock.calls[0][0]).toEqual(fakeNote.id);
      expect(mockNoteCtx.updateNote.mock.calls[0][1]).toMatchObject({
        tags: ["testing", "is", "awesome"],
      });
      expect(getByText("awesome")).toBeInTheDocument();
    });
  });
});

describe("Form page create note", () => {
  it("creates the note on blur of title field", async () => {
    mockNoteCtx.createNote.mockResolvedValue(fakeNote);
    const { getByLabelText, getByText } = await renderCreating();
    fireEvent.change(getByLabelText("Title"), {
      target: { value: "my new note title" },
    });
    await wait(() => {
      expect(getByLabelText("Title")).toHaveValue("my new note title");
    });
    fireEvent.blur(getByLabelText("Title"));
    await wait(() => {
      expect(mockNoteCtx.createNote).toHaveBeenCalled();
      expect(mockNoteCtx.createNote.mock.calls[0][0]).toMatchObject({
        title: "my new note title",
      });
      expect(getByText(fakeNote.title)).toBeInTheDocument();
    });
  });

  it("does not create on blur of a different field", async () => {
    const { getByLabelText } = await renderCreating();
    fireEvent.change(getByLabelText("Note"), {
      target: { value: "# my new note" },
    });
    await wait(() => {
      expect(getByLabelText("Note")).toHaveValue("# my new note");
    });
    fireEvent.blur(getByLabelText("Note"));
    await wait(() => {
      expect(mockNoteCtx.createNote).not.toHaveBeenCalled();
    });
  });

  it("handles uniqueness error for title on create", async () => {
    const err = new Error();
    err.name = "ConstraintError";
    mockNoteCtx.createNote.mockRejectedValue(err);
    const { getByLabelText, getByText } = await renderCreating();
    fireEvent.change(getByLabelText("Title"), {
      target: { value: "duplicate title" },
    });
    await wait(() => {
      expect(getByLabelText("Title")).toHaveValue("duplicate title");
    });
    fireEvent.blur(getByLabelText("Title"));
    await wait(() => {
      expect(mockNoteCtx.createNote).toHaveBeenCalled();
      expect(mockNoteCtx.createNote.mock.calls[0][0]).toMatchObject({
        title: "duplicate title",
      });
      expect(getByText("Title must be unique")).toBeInTheDocument();
    });
  });

  it("handles other errors that have a message set", async () => {
    const err = new Error("Some other error message");
    mockNoteCtx.createNote.mockRejectedValue(err);
    const { getByLabelText, getByText } = await renderCreating();
    fireEvent.change(getByLabelText("Title"), {
      target: { value: "duplicate title" },
    });
    await wait(() => {
      expect(getByLabelText("Title")).toHaveValue("duplicate title");
    });
    fireEvent.blur(getByLabelText("Title"));
    await wait(() => {
      expect(mockNoteCtx.createNote).toHaveBeenCalled();
      expect(getByText("Some other error message")).toBeInTheDocument();
    });
  });

  it("handles other errors with no message set", async () => {
    const err = new Error();
    mockNoteCtx.createNote.mockRejectedValue(err);
    const { getByLabelText, getByText } = await renderCreating();
    fireEvent.change(getByLabelText("Title"), {
      target: { value: "duplicate title" },
    });
    await wait(() => {
      expect(getByLabelText("Title")).toHaveValue("duplicate title");
    });
    fireEvent.blur(getByLabelText("Title"));
    await wait(() => {
      expect(mockNoteCtx.createNote).toHaveBeenCalled();
      expect(
        getByText("Something went wrong creating this note")
      ).toBeInTheDocument();
    });
  });
});
