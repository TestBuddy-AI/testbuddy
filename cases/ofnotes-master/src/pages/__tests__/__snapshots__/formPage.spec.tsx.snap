// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form page navigation matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-tabs-2"
    >
      <div
        class="MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed"
          style="overflow: hidden;"
        >
          <div
            class="MuiTabs-flexContainer MuiTabs-centered"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
              role="tab"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                Write
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                Preview
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="PrivateTabIndicator-root-4 PrivateTabIndicator-colorSecondary-6 MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <hr
        class="MuiDivider-root"
      />
    </div>
    <div
      class="makeStyles-content-3"
      data-testid="form-container"
    >
      <div
        class="makeStyles-root-8"
      >
        <div
          class="makeStyles-root-9"
        >
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-inp-16 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="title"
              id="title-label"
            >
              Title
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                id="title"
                name="title"
                type="text"
                value="test two"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-savedIcon-14"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
                >
                  <span>
                    Title
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-inp-16 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="md"
              id="md-label"
            >
              Note
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
            >
              <textarea
                aria-describedby="md-helper-text"
                aria-invalid="false"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input makeStyles-textarea-10 MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                id="md"
                name="md"
                rows="20"
              >
                # test note 2
              </textarea>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
                >
                  <span>
                    Note
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
              id="md-helper-text"
            >
              Supports github flavored markdown
            </p>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-inp-16 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="tags"
              id="tags-label"
            >
              Tags
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-describedby="tags-helper-text"
                aria-invalid="false"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="tags"
                name="tags"
                type="text"
                value="python"
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
                >
                  <span>
                    Tags
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
              id="tags-helper-text"
            >
              Separated by spaces
            </p>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-delete-12 MuiButton-outlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              Delete
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="makeStyles-root-8"
      >
        <div
          class="makeStyles-wrapper-25"
        >
          <h1
            class="MuiTypography-root makeStyles-title-21 MuiTypography-h3"
          >
            test two
          </h1>
          <div>
            <h1
              class="MuiTypography-root makeStyles-content-22 MuiTypography-h3 MuiTypography-gutterBottom"
              id="test-note-2"
            >
              test note 2
            </h1>
          </div>
          <div
            class="makeStyles-tags-24"
          >
            <div
              class="MuiChip-root makeStyles-chip-23 MuiChip-outlined"
            >
              <span
                class="MuiChip-label"
              >
                python
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
